src/main/java/edu/umich/ctools/esb/utils/WAPI.java

```diff
@@ -0,0 +1,213 @@
+package edu.umich.ctools.esb.utils;
+
+import java.util.HashMap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONObject;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+public class WAPI 
+{
+	private static Log M_log = LogFactory.getLog(WAPI.class);
+
+	private final int HTTP_SUCCESS = 200;
+	private final int HTTP_BAD_REQUEST = 400;
+	private final int HTTP_UNAUTHORIZED = 401;
+	private final int HTTP_FORBIDDEN = 403;
+	private final int HTTP_NOT_FOUND = 404;
+	private final int HTTP_GATEWAY_TMEOUT = 504;
+	private final int HTTP_UNKNOWN_ERROR = 666;
+
+	private final static String CONTENT_TYPE = "Content-Type";
+	private final static String CONTENT_TYPE_PARAMETER = "application/x-www-form-urlencoded";
+	private final static String AUTHORIZATION = "Authorization";
+	private final static String GRANT_TYPE = "grant_type";
+	private final static String CLIENT_CREDENTIALS = "client_credentials";
+	private final static String SCOPE = "scope";
+	private final static String PRODUCTION = "PRODUCTION";
+	private final static String ERROR_MSG = "No results due to error. See meta for more details.";
+
+	private final String BEARER = "Bearer";
+
+	private String apiPrefix;
+	private String tokenServer;
+	private String key;
+	private String secret;
+	private String renewal;
+	private String token;
+	
+	public WAPI() {
+		super();
+	}
+
+	//WAPI constructor will have single variable which will be a map holding all necessary variables
+	//The values for the map should come from a properties file used by the application that depends
+	//on this library. For example, if ccm uses the esb-utils library, the ccm properties should
+	//contain the values for the map being passed in to the WAPI constructor.
+	public WAPI(HashMap<String, String> value) {
+		this.setApiPrefix(value.get("apiPrefix"));
+		this.tokenServer = value.get("tokenServer");
+		this.key = value.get("key");
+		this.secret = value.get("secret");
+		this.renewal = buidRenewal(this.key, this.secret);
+		
+		M_log.info("tokenServer: " + tokenServer);
+		M_log.info("key: " + this.key);
+		M_log.info("secret: " + this.secret);
+		M_log.info("renewal: " + this.renewal);
+		M_log.info("token: " + this.token);
+	}
+	
+	public String getApiPrefix() {
+		return apiPrefix;
+	}
+
+	public void setApiPrefix(String apiPrefix) {
+		this.apiPrefix = apiPrefix;
+	}
+	
+	//a token must be created at the time of construction
+	//this token will allow use of the ESB APIs
+	public String buidRenewal(String key, String secret) {
```

Comment:
RE line 64: Best to print only the first and last couple of characters of the key, secret, and token.  E.g. "e3..f4".  That's enough to confirm which credentials are being used without giving them away.  You'll need to take out anything that prints the secret information later anyway, so it's better not to put in it.